tests/test_other.py

```diff
@@ -7481,7 +7481,7 @@ def test_separate_dwarf_with_filename(self):
     self.assertExists('with_dwarf2.wasm')
     with open(os.path.join('subdir', 'output.wasm'), 'rb') as f:
       wasm = f.read()
-      self.assertIn(b'../with_dwarf2.wasm', wasm)
+      self.assertIn(bytes(os.path.join('..', 'with_dwarf2.wasm'), 'ascii'), wasm)
```

Comment:
Maybe it's better to fix the implementation instead? E.g. for source maps we did the `\` -> `/` replacement right in the implementation, and `external_debug_info` technically is also specced as a relative *URL* not a *path* so it should always use `/` as a delimited.

Reply:
I'd be ok with that, but I have no idea if it would work in tools. Do you know?

Reply:
Or, I guess if what we emit now violates the spec, then it would be a bug if the tools didn't work with this change...?

Reply:
> Or, I guess if what we emit now violates the spec, then it would be a bug if the tools didn't work with this change...?

Well we already had a bug with any relative paths whenever files are not in the same folder. Your previous PR fixed that for Unix-like systems, now we just need to add the `/` replacement for Windows too.

But yeah, it never worked before either.

Reply:
As in, it did work correctly, but only in the common case when files were generated in the same folder, and in that case relative URL is something like "external-info.wasm" so it doesn't care about `/`.

Reply:
TL;DR: yeah it should work with tools, and I think test as it stands now catches a real bug.

Reply:
Interesting, makes sense.

We may want to land this meanwhile to fix CI, but I agree we should fix it properly.

For fixing it properly I am a little unclear on how to do that. There isn't a general way to convert a windows path into a URL, is there? Like if the path starts with `C:\` (which the user can provide on `-gseparate-dwarf=..`). Maybe we should error on such things, but I'm not sure how to even check for them, offhand (like, what is a valid windows drive prefix, etc.).

Reply:
Also, we are calling python's relpath here, which may do windows-specific things too, which we may need to undo. Or, we could write our own relative path handler..?

Reply:
> Like if the path starts with C:\ 

You're already doing the `relpath`, so in most cases this shouldn't matter. For now I'd suggest to just do what we did for sourceMapURL where we ran into exactly same situation. Here's the path normalization code: https://github.com/emscripten-core/emscripten/blob/master/tools/wasm-sourcemap.py#L252

Reply:
The only case it doesn't work for is when files are generated on different drives (https://github.com/emscripten-core/emscripten/issues/10154), but it works for other 95% of cases and for now should be good enough.

Reply:
So, another tl;dr - just replacing `\` with `/` should be good enough for this fix.

Reply:
It seems like relpath, replacing `os.pathsep` with `/` should work? I assume there's no way relpath can produce a path with a drive letter or anything other than just `..`?

Reply:
> It seems like relpath, replacing `os.pathsep` with `/` should work?

Yep, that's what I'm trying to say in my last TLDR, sorry if still unclear :)

Reply:
No, you were clear, I was just adding that that would be my understanding too, modulo any surprises from reldir...

Reply:
Sounds good, thanks! Done.

Yeah, makes sense that if it's good enough there, it should be good enough here.